File: 백준/Bronze/9506. 약수들의 합/약수들의 합.cc
#include <stdio.h>
int main() {
    int a;
    
    while(1) {
        scanf("%d", &a);
        if(a == -1) break;
    int arr[a];
    int idx = 0, sum = 0;
    for (int i = 1; i < a; i++)
    {
        if(a%i == 0) {
            arr[idx++] = i;
            sum+=i;
        }
    }

    if(sum == a) {
        printf("%d = %d ", a, arr[0]);
        for(int i =1; i<idx; i++) {
            printf("+ %d ", arr[i]);
        }
    }
    else {
        printf("%d is NOT perfect.\n", a);
    }
    }
    return 0;
}

File: 백준/Bronze/9506. 약수들의 합/약수들의 합.py
while True:
  n = int(input())
  if n == -1:
    break
  li = []
  for i in range(1, n):
    if n % i == 0:
      li.append(i)
  if sum(li) == n:
    print(n, end=' = ')
    for i in range(len(li)):
      print(li[i], end=' ')
      if i + 1 != len(li):
        print('+', end=' ')
    print()
  else:
    print(f'{n} is NOT perfect.')